x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -83,7 +84,24 @@ interface IWaterfallItemSpan extends IWaterfallItemBase {
   docType: 'span';
 }
 
-export type IWaterfallItem = IWaterfallItemSpan | IWaterfallItemTransaction;
+export interface IWaterfallItemAgentMark
+  extends Pick<IWaterfallItemBase, 'id' | 'name' | 'offset'> {
+  docType: 'agentMark';
+}
+
+export interface IWaterfallItemError extends Omit<IWaterfallItemBase, 'name'> {
+  error: APMError;
+  docType: 'error';
+  message?: string;
+  serviceColor?: string;
+}
+
+type IWaterfallTraceItem =
+  | IWaterfallItemSpan
+  | IWaterfallItemTransaction
+  | IWaterfallItemError;
```

Comment:
@sqren WDYT about this? I created the Trace type, which consists of `span`, `transaction` and `error`, to avoid having to check if the type is different than `agentMark`, as it doesn't have most of the properties from `IWaterfallItemBase`

Reply:
I don't like this implementation, removing it!!! 😅